src/generate_trigger.ts

```diff
@@ -0,0 +1,112 @@
+import { parse } from "./deps.ts";
+
+export type WorkflowRecord = {
+  "callback_id": string;
+};
+
+interface Inputs {
+  [key: string]: { value: string };
+}
+
+export type TriggerFileRecord = {
+  "type": string;
+  "name": string;
+  "description"?: string;
+  "workflow": WorkflowRecord;
+  "inputs"?: Inputs;
+};
+
+export type TriggersPayload = {
+  triggers: TriggerFileRecord[];
+};
+
+interface Response {
+  ok: boolean;
+  files: Array<string>;
+  error?: {
+    message: string;
+  } | null;
+}
+
+export const generateTriggerFile = async (
+  input: string[],
+): Promise<Response> => {
+  // Parse JSON payload
+  const source = parse(input).source as string;
+  if (!source) throw new Error("A source object needs to be defined");
+  const payload: TriggersPayload = JSON.parse(source);
+
+  // Create the triggers directory if it does not exist
+  const directory = "triggers";
+  try {
+    await Deno.stat(directory);
+  } catch (error) {
+    if (error instanceof Deno.errors.NotFound) { // create directory
+      await Deno.mkdir(directory);
+    } else {
+      throw error;
+    }
+  }
+
+  // Generate the files
+  const files: Array<string> = [];
+  await Promise.all(payload.triggers.map(async (tr: TriggerFileRecord) => {
+    const templateString = SlackTriggerTemplate(tr);
+    const filename = directory + `/${tr.workflow.callback_id}.ts`;
```

Comment:
nit: append `_trigger`
```suggestion
    const filename = directory + `/${tr.workflow.callback_id}_trigger.ts`;
```

Reply:
I thought the organization of triggers into a 'triggers/' folder (and other components into their respective workflows/ and functions/ folders) was our standard. Does that only apply to how we organize our own templates?

I recall a thread where not including '<>_trigger' / '<>_workflow' etc. was proposed to be less repetitive (as it was assumed the file would be organized like 'triggers/<triggerIdentifier>.ts'); now I'm having trouble finding that thread again...

If we support generating the file anywhere, appending '_trigger' to the filename makes sense!

Would we not want to place the file according to our own standard though? :confused-parrot:

Reply:
It seems like [deno-hello-world](https://github.com/slack-samples/deno-hello-world/tree/main/triggers) appends `_trigger` and `_workflow`. I don't have a super strong opinion outside of what you said — if we allow generation of the file anywhere, having `_trigger` is important.